NorthstarDedicatedTest/plugins.cpp

```diff
@@ -0,0 +1,445 @@
+SQRESULT SQ_UpdateGameStateClient(void* sqvm)
+{
+	AcquireSRWLockExclusive(&gameStateLock);
+	AcquireSRWLockExclusive(&serverInfoLock);
+	gameState.players = ClientSq_getinteger(sqvm, 1);
+	gameState.ourScore = ClientSq_getinteger(sqvm, 2);
+	gameState.secondHighestScore = ClientSq_getinteger(sqvm, 3);
+	gameState.highestScore = ClientSq_getinteger(sqvm, 4);
+	serverInfo.roundBased = ClientSq_getbool(sqvm, 5);
+	serverInfo.scoreLimit = ClientSq_getbool(sqvm, 6);
+	ReleaseSRWLockExclusive(&gameStateLock);
+	ReleaseSRWLockExclusive(&serverInfoLock);
+	return SQRESULT_NOTNULL;
+}
+
+// string id, string name, string password, int players, int maxPlayers, string map, string mapDisplayName, string playlist, string
+// playlistDisplayName
+SQRESULT SQ_UpdateServerInfo(void* sqvm)
+{
+	AcquireSRWLockExclusive(&gameStateLock);
+	AcquireSRWLockExclusive(&serverInfoLock);
+	serverInfo.id = ClientSq_getstring(sqvm, 1);
+	serverInfo.name = ClientSq_getstring(sqvm, 2);
+	serverInfo.password = ClientSq_getstring(sqvm, 3);
+	gameState.players = ClientSq_getinteger(sqvm, 4);
+	serverInfo.maxPlayers = ClientSq_getinteger(sqvm, 5);
+	gameState.map = ClientSq_getstring(sqvm, 6);
+	gameState.mapDisplayName = ClientSq_getstring(sqvm, 7);
+	gameState.playlist = ClientSq_getstring(sqvm, 8);
+	gameState.playlistDisplayName = ClientSq_getstring(sqvm, 9);
+	ReleaseSRWLockExclusive(&gameStateLock);
+	ReleaseSRWLockExclusive(&serverInfoLock);
+	return SQRESULT_NOTNULL;
+}
+
+// int maxPlayers
+SQRESULT SQ_UpdateServerInfoBetweenRounds(void* sqvm)
+{
+	AcquireSRWLockExclusive(&serverInfoLock);
+	serverInfo.id = ClientSq_getstring(sqvm, 1);
+	serverInfo.name = ClientSq_getstring(sqvm, 2);
+	serverInfo.password = ClientSq_getstring(sqvm, 3);
+	serverInfo.maxPlayers = ClientSq_getinteger(sqvm, 4);
+	ReleaseSRWLockExclusive(&serverInfoLock);
+	return SQRESULT_NOTNULL;
+}
+
+// float timeInFuture
+SQRESULT SQ_UpdateTimeInfo(void* sqvm)
+{
+	AcquireSRWLockExclusive(&serverInfoLock);
+	int endTimeFromNow = ceil(ClientSq_getfloat(sqvm, 1));
+	const auto p1 = std::chrono::system_clock::now().time_since_epoch();
+	serverInfo.endTime = std::chrono::duration_cast<std::chrono::seconds>(p1).count() + endTimeFromNow;
+	ReleaseSRWLockExclusive(&serverInfoLock);
+	return SQRESULT_NOTNULL;
+}
+
+// bool loading
+SQRESULT SQ_SetConnected(void* sqvm)
+{
+	AcquireSRWLockExclusive(&gameStateLock);
+	gameState.loading = ClientSq_getbool(sqvm, 1);
+	ReleaseSRWLockExclusive(&gameStateLock);
+	return SQRESULT_NOTNULL;
+}
+
+SQRESULT SQ_UpdateListenServer(void* sqvm)
+{
+	AcquireSRWLockExclusive(&serverInfoLock);
+	serverInfo.id = g_MasterServerManager->m_ownServerId;
+	serverInfo.password = Cvar_ns_server_password->GetString();
+	ReleaseSRWLockExclusive(&serverInfoLock);
+	return SQRESULT_NOTNULL;
+}
+
+int getServerInfoChar(char* out_buf, size_t out_buf_len, ServerInfoType var)
+{
+	/* if (var > ...)
+	{
+		return -1; // no such var
+	}*/
```

Comment:
Maybe remove these comments if we're just checking it in the switch statement anyways.

Reply:
Forgot to remove this